### PR TITLE
Update cypher differences

### PR DESCRIPTION
**Pull request by katarinasupe**
### Description

I updated the Cypher differences page with new information, workarounds and examples. 

### Pull request type

Please check what kind of PR this is:

- [x] Fix or improvement of an existing page

### Related PRs and issues

Closes https://github.com/memgraph/documentation/issues/100

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

File: pages/querying/differences-in-cypher-implementations.mdx [hunk 1]
```diff
@@ -0,0 +1,307 @@
+---
+title: Differences in Cypher implementation
+description: Compare and contrast different Cypher implementations in Memgraph. Comprehensive documentation to level up your graph computing expertise.
```
Comment (antejavor) [inline pages/querying/differences-in-cypher-implementations.mdx:3]:
```suggestion
description: Compare and contrast different Cypher implementations in Memgraph. 
```

Comment (antejavor) [reply pages/querying/differences-in-cypher-implementations.mdx:3]: Sounds auto-generated, not sure is it intentional :D

File: pages/querying/differences-in-cypher-implementations.mdx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+MATCH (p:Person {name:'Tom Hanks'})-[r]->(m:Movie)
+WHERE type(r) != "ACTED_IN"
+RETURN type(r) AS type, m.title AS movies;
+```
+
+### Search for patterns of a fixed length
+
+In Neo4j, to search for patterns of a fixed length, you would use the following construct:
+```cypher
+MATCH (tom:Person {name:'Tom Hanks'})--{2}(colleagues:Person)
```
Comment (antejavor) [inline pages/querying/differences-in-cypher-implementations.mdx:134]: Shoud `{2}` be in between `--`?

File: pages/querying/differences-in-cypher-implementations.mdx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+{<h4>EXISTS subqueries</h4>}
+Such clause is not supported in Memgraph, but you can use `exists()` [pattern function](/querying/functions#pattern-functions) with the `WHERE` clause to [filter with pattern expressions](/querying/clauses/where#17-filter-with-pattern-expressions). 
+
+The following constructs are not yet supported in Memgraph:
+- **`EXISTS` subquery with `WHERE` clause** -> Track progress on [GitHub](https://github.com/memgraph/memgraph/issues/1483) and add a comment if you require such a feature.
+- **Nesting `EXISTS` subqueries** -> Track progress on [GitHub](https://github.com/memgraph/memgraph/issues/905) and add a comment if you require such a feature.
+- **`EXISTS` subquery outside of a `WHERE` clause** -> Track progress on [GitHub](https://github.com/memgraph/memgraph/issues/1308) and add a comment if you require such a feature.
+
+For all other unsupported constructs that you require, please open an issue on our [GitHub repository](https://github.com/memgraph/memgraph/issues). Learn more in our Community Call on [Subqueries & Patterns in Filtering Clauses](https://www.youtube.com/watch?v=QYFU5d_xLIs&list=PL7Eotag2rRhZssS4f11PKAHuCykMCljg3).
```
Comment (antejavor) [inline pages/querying/differences-in-cypher-implementations.mdx:175]: Maybe this can be general note at the end also?

**Review by antejavor — APPROVED** [review]

File: pages/querying/differences-in-cypher-implementations.mdx (hunk 2 above)
Comment (katarinasupe) [reply pages/querying/differences-in-cypher-implementations.mdx:134]: Believe it or not, no 😂

File: pages/querying/differences-in-cypher-implementations.mdx (hunk 3 above)
Comment (katarinasupe) [reply pages/querying/differences-in-cypher-implementations.mdx:175]: I agree, I'll add it

File: pages/querying/differences-in-cypher-implementations.mdx (hunk 1 above)
Comment (katarinasupe) [reply pages/querying/differences-in-cypher-implementations.mdx:3]: TBH, didn't even look at it, it's old. I agree, will update it

**Comment by katarinasupe** [comment]
@kgolubic after your review, this can be merged.

**Review by kgolubic — APPROVED** [review]
Everything looks ok.

**Pull request merged**